docs/_includes/Modules.md

```diff
@@ -648,6 +649,75 @@ Ordinarily a module must be _imported_ in order for its constitutent
 declarations to be visible inside a given module `M`. However, for the
 resolution of qualified names this is not the case.
 
+This example shows that the resolution of the refinement parent does not
+use any local names:
+```dafny
+{% include Example-Refines1.dfy %}
+```
+In the example, the `A` in `refines A` refers to the global `A`, not the submodule `A`.
+
+
+A module is a collection of declarations, each of which has a name.
+These names are held in two namespaces.
+
+* The names of export sets
+* The names of all other declarations, including submodules and aliased modules
+
+In addition names can be classified as _local_ or _imported_.
+
+* Local declarations of a module are the declarations
+ that are explicit in the module and the
+local declarations of the refinement parent. This includes, for
+example, the `N` of `import N = ` in the refinement parent, recursively.
+* Imported names of a module are those brought in by `import opened` plus
+the imported names in the refinement parent.
+
+Within each namespace, the local names are unique. Thus a module may
+not reuse a name that a refinement parent has declared (unless it is a
+refining declaration, which replaces both declarations, as described
+[here](#TODO)).
+
+Local names take precedence over imported names. If a name is used more than
+once among imported names (coming from different imports), then it is
+ambiguous to _use_ the name without qualification.
```

Comment:
unless they refer to types and those types are equal.

Reply:
Right